.github/workflows/PR-build.yml

```diff
@@ -170,31 +200,35 @@ jobs:
           
   run-test-case:
```

Comment:
Taking a step back, is it worth running these on every PR build? I think it's pretty conventional to have integration type of tests only running on main build since unit tests should have good coverage anyways. To reduce the overhead of adding if-statement complexity to so many steps, I think it's good to rethink this file from the ground up.

For PR builds, I think just the current `build` job is enough. And as long as go modules / tests are cached using actions cache there probably isn't a big reason to skip by path if it's the only job.

Reply:
We're only testing a small subset of the integration tests. Even if we're only doing the `build` job, I still think it's worth skipping it. There's no benefit to the build.